bin/aursearch

```diff
@@ -164,15 +160,12 @@ if ((!$#)); then
     exit 1
 fi
 
-case $searchby in
-  "name-desc") urls=("$*") ;;&
-  "name"|"maintainer") urls=("$@") ;;&
-  "name"|"maintainer"|"name-desc")
-    { for u in "${urls[@]}"; do
-      echo "$aurweb/rpc/?v=5&type=search&by=$searchby&arg=${u/ /%20}"
-    done } | getjson | parse "$format"
-  ;;
-  *) match "$list" "$@" | split | getjson | parse "$format"
-esac
+if [ -n "$searchby" ]; then
+  { for q in "$@"; do
+    echo "$aurweb/rpc/?v=5&type=search&by=$searchby&arg=${q/ /%20}"
```

Comment:
I'd suggest extracting urlencoding from split as another step in the pipeline. The search method in the rpc doesn't support passing an array of arguments so each search term requires a separate request.